core/src/main/java/org/apache/calcite/rel/metadata/RelMdUtil.java

```diff
@@ -831,6 +831,9 @@ public static double getMinusRowCount(RelMetadataQuery mq, Minus minus) {
     }
     double innerRowCount = left * right * selectivity;
     switch (join.getJoinType()) {
+    case ASOF:
+    case LEFT_ASOF:
```

Comment:
Is ASOF similar to inner join or left join?

Reply:
In this implementation we have both ASOF and LEFT ASOF.
ASOF itself is an INNER join.

Reply:
I agree that LEFT_ASOF returns left. Would it be more accurate for asof to return (left * selectivity)?

Reply:
I am not familiar with this part of the code, I see that other places use 1 - selectivity.

Reply:
My understanding is that for a regular inner join, selectivity means the probability of the condition being true in the result, and 1 - selectivity is because for an outer join we need to add the unmatched rows on the null side.
For asof join, we should consider how many matches there are against the left table, and we also could to consider the matchCondition, which is more complicated. Perhaps this optimization can be discussed in another issue.